modules/pam_xauth/pam_xauth.c

```diff
@@ -685,13 +704,17 @@ pam_sm_open_session (pam_handle_t *pamh, int flags UNUSED,
 			goto cleanup;
 		}
 
+		if (debug) {
+			pam_syslog(pamh, LOG_DEBUG, "set environment variable '%s'",
+				   xauthority);
+		}
 		/* Set the new variable in the environment. */
 		if (pam_putenv (pamh, xauthority) != PAM_SUCCESS)
 			pam_syslog(pamh, LOG_ERR,
 				   "can't set environment variable '%s'",
 				   xauthority);
-		putenv (xauthority); /* The environment owns this string now. */
-		xauthority = NULL; /* Don't free environment variables. */
+		if (putenv (xauthority) == 0) /* The environment owns this string now. */
+			xauthority = NULL; /* Don't free environment variables. */
```

Comment:
What it has to do with NFS and the cookiefile?

Reply:
Just a security about the variables used and memory usage. Has been removed from new commits